### PR TITLE
cryptodev-linux: Update to 1.10

### PR DESCRIPTION
**Pull request by diizzyy**
Update cryptodev-linux to 1.10
Switch from git to codeload generated tarball

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by diizzyy** [comment]
Compile tested: mvebu, Linksys WRT3200ACM, OpenWrt master
Runtime tested: mvebu, Linksys WRT3200ACM, OpenWrt master

**Comment by diizzyy** [comment]
@Ansuel

**Comment by Ansuel** [comment]
ok for me 
Why switch to codeload?

**Comment by diizzyy** [comment]
I don't see the the benefit of using git when we don't need to, perhaps I'm missing something. I'd rather use a release tarball but there isn't one available.

File: package/kernel/cryptodev-linux/Makefile [hunk 1]
```diff
@@ -10,19 +10,18 @@ include $(TOPDIR)/rules.mk
...
+PKG_SOURCE_URL:=https://codeload.github.com/$(PKG_NAME)/$(PKG_NAME)/tar.gz/$(PKG_NAME)-$(PKG_VERSION)?
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_HASH:=833ab7c5c88d2b700a7c702a151254c089a3058886a63cc7d12630e364b8ea83
 PKG_LICENSE:=GPL-2.0
 PKG_LICENSE_FILES:=COPYING
 
 PKG_MAINTAINER:=Ansuel Smith <ansuelsmth@gmail.com>
 
-PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+ PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/$(PKG_NAME)-$(PKG_NAME)-$(PKG_VERSION)
```
Comment (dedeckeh) [inline package/kernel/cryptodev-linux/Makefile]: Why twice $PKG_NAME in $PKG_BUILD_DIR ?

Comment (diizzyy) [reply package/kernel/cryptodev-linux/Makefile]: Because GitHub adds the "username" which happens to the be same as repo name in this case

Comment (ynezz) [reply package/kernel/cryptodev-linux/Makefile]: @diizzyy BTW, you've introduced one space indent before `PKG_BUILD_DIR`

Comment (diizzyy) [reply package/kernel/cryptodev-linux/Makefile]: Fixed, thanks for point that out :-)

**Comment by escalade** [comment]
There are release tarballs available: https://github.com/cryptodev-linux/cryptodev-linux/releases

The benefit of using git is that you can add features that aren't in a release yet.

**Comment by neheb** [comment]
@cshoredaniel My reasons are:
Simpler Makefile
Fewer buildbot failures regarding local generation of tarballs

Codeload cannot be used when a git repository has submodules as they end up missing.
Codeload also only generates .tar.gz files instead of the smaller .tar.xz.

@jow- also has questions regarding the actual tarballs themselves as they use a weird format.

A few guidelines regarding this can be found here: https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#package-sources-archives-and-repositories

**Comment by diizzyy** [comment]
@dedeckeh 
Can we merge this please?

**Comment by dedeckeh** [comment]
@Ansuel As package maintainer is the PR ok for you ?

**Comment by Ansuel** [comment]
fine for me as i already said

**Comment by dedeckeh** [comment]
Patch pushed to master (https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=e4088cb84be4c808d79554d098d2b3e4b8f1608d); thx

**Pull request closed without merging**